src/qgis_plugin_dev_tools/build/packaging.py

```diff
@@ -54,20 +66,31 @@ def copy_runtime_requirements(
     if len(dev_tools_config.runtime_distributions) > 0:
         vendor_path = build_directory_path / plugin_package_name / "_vendor"
         vendor_path.mkdir(parents=True)
-        (vendor_path / "__init__.py").touch()
+        vendor_init_file = vendor_path / "__init__.py"
+        vendor_init_file.touch()
+        if dev_tools_config.append_distributions_to_path:
+            vendor_init_file.write_text(VENDOR_PATH_APPEND_SCRIPT)
 
     runtime_package_names = []
-    # copy dist infos (licenses etc) and all provided top level packages
+    # copy dist infos (licenses etc.) and all provided top level packages
     for dist in dev_tools_config.runtime_distributions:
         dist_info_path = Path(dist._path)  # type: ignore
+        if Path(sys.base_prefix) in dist_info_path.parent.parents:
```

Comment:
maybe this could also check for "if dist name not in explicit runtime requirements", meaning that only those requirements that come from auto-recursive logic can be omitted if provided in the base environment where the build is run in

this could allow to force-bundle some dependency thats provided by qgis env also

Reply:
Good idea!